cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceUtils.java

```diff
@@ -144,4 +146,46 @@ public static void updateKafkaAutoRebalanceStatus(KafkaStatus kafkaStatus, Kafka
             }
         }
     }
+
+    /**
+     * Updates the {@code KafkaRebalanceStatus} with a new Warning condition based
+     * on the provided {@link Throwable}. If a Warning condition with the same reason
+     * and message already exists, no update is performed.
+     *
+     * @param status    The {@link KafkaRebalanceStatus} object whose conditions will be updated.
+     * @param exception The {@link Throwable} containing the reason and message for the Warning condition.
+     */
+    public static void addWarningCondition(KafkaRebalanceStatus status, Throwable exception) {
```

Comment:
Is there not already a generic menthod for this that will work on classes derived from `Status`?

Reply:
We do have some generic methods as part of the `Status` class that can add a Condition to the `Status`'s condition list we need a little extra logic wrapped around to not replace existing a Warning Condition w/ same reason + message.